### PR TITLE
bonding related changes

### PR DESCRIPTION
**Pull request by mbpavan**
# Description

Bonded related config documentation update and minor change to have an option to create service to create bond files.
Fixes # (issue)

## Type of change

Please select the appropriate options:

New minor addition  (non-breaking change which adds functionality)
- This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
Install and upgrades have been tested extensively.
- Versions: 
- 4.4.5.
- Hardware:
HP 
## Checklist

- [I have performed a self-review of my own code

### CODE REVIEW

**Comment by DirectedSoul1** [comment]
Hi, @mbpavan  thanks for raising this PR. I have added some comments for some of the sections.

File: documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc [hunk 1]
```diff
@@ -0,0 +1,128 @@
+[id="ansible-playbook-create-bonded-interfaces-using-script.adoc"]
+
+= Create bonded interfaces
+
+If you need to create a bonded provisioning/baremetal interface, the ifcfg files
+can be placed under the "fake" roots for the masters and workers under the
+`roles/installer/files/customize_filesystem/{master,worker}` directories as shown in the above section 4.7
```
Comment (DirectedSoul1) [inline documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc:7]: Is this targetted for `4.7` or later or is it only tested on `4.4.5` as mentioned?

File: documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+This can also be done by creating a script /usr/local/bin/create-ifcfg-files.sh under 
+`roles/installer/files/customize_filesystem/{master,worker}`
+and by setting by setting variable create_bond_service="true" to create ifcfg files.
```
Comment (DirectedSoul1) [inline documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc:11]: here `by setting` is repeated twice, maybe a typo?

File: documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+# bond1 creation
+cat <<EOF > /etc/sysconfig/network-scripts/ifcfg-bond1
+BONDING_OPTS=mode=active-backup
+TYPE=Bond
+BONDING_MASTER=yes
+PROXY_METHOD=none
+ONBOOT=yes
+BOOTPROTO=dhcp
+DHCP_CLIENT_ID=$(ip link show dev ens3f0 | awk '/link/ {print $2}')
```
Comment (DirectedSoul1) [inline documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc:44]: The `dev` name `ens3f0` seems like environment specific can this be generalized?

File: documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+IPV4_DHCP_TIMEOUT=2147483647
+NAME=bond1
+DEVICE=bond1
+ONBOOT=yes
+EOF
+
+# bond1 members creation
+cat <<EOF > /etc/sysconfig/network-scripts/ifcfg-ens3f0
+TYPE=Ethernet
+DEVICE=ens3f0
```
Comment (DirectedSoul1) [inline documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc:54]: can this `ens3f0` be generalized?

File: documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+SLAVE=yes
+EOF
+
+# bond0 creation
+cat <<EOF > /etc/sysconfig/network-scripts/ifcfg-bond0
+DEVICE=bond0
+BONDING_OPTS=mode=active-backup
+TYPE=Bond
+BOOTPROTO=dhcp
+DHCP_CLIENT_ID=$(ip link show dev eno5np0 | awk '/link/ {print $2}')
```
Comment (DirectedSoul1) [inline documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc:76]: the `dev` name `eno5np0` seems like environment-specific, can this be generalized?

**Comment by mbpavan** [comment]
> Hi, @mbpavan thanks for raising this PR. I have added some comments for some of the sections.
> Also, is there a chance that we can make this as a `day-2` ops via MCO, instead of updating/modifying the `ifcfg` files?
   There is already support for day 2 bonding. This PR doc/minor task to make it day 0. The bonding will come up on first install.
> HW used is `HP`; any idea of this working on `DELL` h/w as well?
I haven't tried it on DELL HW as I did not have. Please make sure there are 2 provisioning and/or 2 bare-metal in case of testing
bond creations

File: documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc (hunk 1 above)
Comment (mbpavan) [reply documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc:7]: This is only tested on 4.4.5 and upgrades to 4.5.x. This has not been tested on 4.6 or 4.7

File: documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc (hunk 2 above)
Comment (mbpavan) [reply documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc]: done

File: documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc (hunk 3 above)
Comment (mbpavan) [reply documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc]: done

File: documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc (hunk 4 above)
Comment (mbpavan) [reply documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc]: done

File: documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc (hunk 5 above)
Comment (mbpavan) [reply documentation/ansible-playbook/modules/ansible-playbook-create-bonded-interfaces-using-script.adoc]: done

**Comment by PavanBheeman** [comment]
Hello!

Any update on merging this or any comments on the same?

Thanks,
Pavan

**Comment by DirectedSoul1** [comment]
Hi @mbpavan these changes as of now will be on `hold` for the repo, and will be updated/changed/reviewed from 4.8+ releases. 
Also, is it still valid for versions above `4.4.5` ?

**Pull request closed without merging**